drivers/soundwire/stream.c

```diff
@@ -2134,9 +2136,19 @@ int sdw_stream_add_slave(struct sdw_slave *slave,
 	struct sdw_master_runtime *m_rt;
 	bool alloc_master_rt = false;
 	bool alloc_slave_rt = false;
+	unsigned long time;
 
 	int ret;
 
+	time = wait_for_completion_timeout(&slave->initialization_complete,
+					   msecs_to_jiffies(CODEC_INIT_TIMEOUT));
+	if (!time) {
+		dev_err(&slave->dev, "%s: Initialization not complete, timed out\n", __func__);
+		sdw_show_ping_status(slave->bus, true);
+
+		return -ETIMEDOUT;
+	}
```

Comment:
Definitely have some concerns about moving this to the core, could we do this in the hw_params of the affected drivers instead? Not all drivers use this initialization complete stuff, and this will have impacts across everything.